src/main/java/reposense/model/CliRunConfiguration.java

```diff
@@ -0,0 +1,68 @@
+package reposense.model;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import reposense.parser.AuthorConfigCsvParser;
+import reposense.parser.GroupConfigCsvParser;
+import reposense.parser.InvalidCsvException;
+import reposense.parser.InvalidHeaderException;
+import reposense.parser.RepoConfigCsvParser;
+import reposense.system.LogsManager;
+
+/**
+ * Represents RepoSense run configured by CLI.
+ */
+public class CliRunConfiguration implements RunConfiguration {
+    private static final Logger logger = LogsManager.getLogger(CliRunConfiguration.class);
+
+    private final CliArguments cliArguments;
+
+    public CliRunConfiguration(CliArguments cliArguments) {
+        this.cliArguments = cliArguments;
+    }
+
+    /**
+     * Constructs a list of {@link RepoConfiguration}.
+     *
+     * @throws IOException if user-supplied csv file does not exist or is not readable.
+     * @throws InvalidCsvException if user-supplied repo-config csv is malformed.
+     * @throws InvalidHeaderException if user-supplied csv file has header that cannot be parsed.
+     */
+    @Override
+    public List<RepoConfiguration> getRepoConfigurations()
```

Comment:
Really like how with this change, this method and its variants is no longer defined in the main reposense file